src/AppCard.test.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { render } from 'react-testing-library'
+
+import electionSample from './data/electionSample.json'
+
+import Root, { App } from './App'
+import { ActivationCardData, AdminCardData, Election } from './config/types'
+
+const election = electionSample as Election
+
+beforeEach(() => {
+  window.localStorage.clear()
+  window.location.href = '/'
+})
+
+it(`App fetches the card data every 1 second`, async () => {
+  fetchMock.resetMocks()
+  jest.useFakeTimers()
+
+  fetchMock.mockResponses(['', {}], ['', {}], ['', {}])
+
+  // load the sample election
+  window.location.href = '/#sample'
+  render(<Root />)
+
+  expect(window.setInterval).toHaveBeenCalledTimes(1)
+
+  jest.advanceTimersByTime(3000)
+
+  expect(fetchMock.mock.calls.length).toEqual(3)
+  expect(fetchMock.mock.calls).toEqual([
+    ['/card/read'],
+    ['/card/read'],
+    ['/card/read'],
+  ])
+
+  jest.useRealTimers()
+})
+
+it(`CardData processing processes card data properly`, () => {
+  const div = document.createElement('div')
+  // @ts-ignore - App expects ReactRouter props, but are unnecessary for this test.
+  const app = (ReactDOM.render(<App />, div) as unknown) as App
```

Comment:
I really wanted to find a better way to do this, but was unable to do it within my self-imposed time box. 😐

Reply:
Yeah, I tried a lot of different things, too.